### DISCUSSION TITLE
IModel not present in 7.0.0-alpha2

### DISCUSSION THREAD

**Discussion by himanshuz2** [rabbitmq/rabbitmq-dotnet-client#1456] [Q&A] [answered]
### Describe the bug

This breaks backward compatibility.  Interfaces should not be removed when updating packages. 

### Reproduction steps

Update nuget to RabbitMQ.Client 7.0.0-alpha2.  Existing code such as this  
private IModel _channel = null;
private IModel _consumerChannel = null;
break.

### Expected behavior

 Interfaces should not be removed when updating packages. 

### Additional context

I want to begin using the async functionality but I don't understand why renaming or dropping IModel Interface.

**Comment by lukebakken** [comment, marked as answer]
This library follows Semantic Versioning.

Version 7 increments the major version of this library, which allows backwards-incompatible changes.

Please see this documentation: https://semver.org/#summary

`IModel` has been renamed to `IChannel` to be in-line with the AMQP entity name, as well as other AMQP client libraries.

**Comment by michaelklishin** [comment]
When you are upgrading to an alpha of a major version, you **should** expect breaking changes, in particular in more mature libraries, because that's the only opportunity for its maintainers to introduce any breaking changes.

`IModel` was an unfortunately picked name from 2007. No other client uses it. This client should not either.

**Reply by himanshuz2** [reply]
Ok, I get it. Is there any documentation for breaking changes? Another change is IModel.CreateProperties method is not present in the new Interface.  What is the replacement for that when using IChannel?

**Reply by michaelklishin** [reply]
7.0 is in its alpha release days. Are you sure you need to use 7.0 if you are not interested in digging in the code and PRs? If not then perhaps wait for it to become at least an RC. By then there will be release notes, just like with any other release.

**Reply by michaelklishin** [reply]
``` csharp
new BasicProperties()
```

is used in many tests. This was the case long before https://github.com/rabbitmq/rabbitmq-dotnet-client/commit/de0a0b4e734bb3f153a188bf7d17e258aaee5667.

A quick search suggests that the `BasicProperties` interface was revisited two years ago in https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1096/.

**Reply by himanshuz2** [reply]
ok. Thanks. Will wait for RC.

**Reply by lukebakken** [reply]
Please continue testing the alpha release if you can. The test suite is very thorough if you need an example of the "new way" to do things. Thanks.
